LFP_analysis/Laser_LFP_Parse_Final.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Jul  6 10:06:32 2017
+
+@author: Bradly
+"""
+#Import necessary tools
+import numpy as np
+import easygui
+import tables
+import os
+
+#Get name of directory where the data files and hdf5 file sits, and change to that directory for processing
+dir_name = easygui.diropenbox()
+os.chdir(dir_name)
+
+#Look for the hdf5 file in the directory
+file_list = os.listdir('./')
+hdf5_name = ''
+for files in file_list:
+	if files[-2:] == 'h5':
+		hdf5_name = files
+
+#Open the hdf5 file
+hf5 = tables.open_file(hdf5_name, 'r+')
+
+# Create a LFP_laser group in the hdf5 file (if one exists, remove and create new)
+try:
+    hf5.remove_node('/LFP_Lasers', recursive = True)
+except:
+    pass
+hf5.create_group('/', 'LFP_Lasers')
+
+#Get laser conditions from hdf5 file (after ancillary analysis.py)
+laser_conditions = hf5.root.ancillary_analysis.laser_combination_d_l[:]
+laser_trials = hf5.root.ancillary_analysis.trials[:]
+
+#Get taste and LFP information
+num_tastes = len(hf5.list_nodes('/spike_trains'))
+lfp_nodes = hf5.list_nodes('/Parsed_LFP')
+
+#Get trial information based on tastants (assumes that each taste has equal number of trials)
+num_trials = hf5.root.spike_trains.dig_in_0.spike_array[:].shape[0]
+
+# Run through the tastes and laser conditions, and build arrays with respective data
+for x in range(laser_conditions.shape[0]):
+    
+    #Create identifier based on laser combination information and create group within "LFP_Lasers" node
+    las_type = 'laser_combos_'+str(int(laser_conditions[x,0]))+'_'+str(int(laser_conditions[x,1]))         
+    hf5.create_group('/LFP_Lasers', str.split(las_type, '/')[-1])
+    
+    #Loop through taste arrays (dig_in files), identify trial number pertaining to laser condition, and built array with respective LFP data
+    for y in range(num_tastes):
+        laser_trial_LFPs = []
+        taste = 'taste_' + str(y)
+        
+        #Target corresponding LFP array for data indexing
+        lfp_coll = np.mean(lfp_nodes[y][:],axis=(0))

```

Comment:
Also, if you are taking the mean only along one axis, you do not have to make the axis number a tuple (by putting it into brackets). This will do:
```
lfp_coll = np.mean(lfp_nodes[y][:],axis=0)
```